soroban-env-host/src/budget.rs

```diff
@@ -364,6 +369,24 @@ impl Budget {
         })
     }
 
+    pub fn with_free_budget<F>(&self, f: F) -> Result<(), HostError>
+    where
+        F: FnOnce() -> Result<(), HostError>,
+    {
+        self.mut_budget(|mut b| {
+            b.enabled = false;
+            Ok(())
+        })?;
+
+        let res = f();
+
+        self.mut_budget(|mut b| {
+            b.enabled = true;
+            Ok(())
+        })?;
+        res
+    }
+
```

Comment:
I think this can be reduced a bit further?

```
        self.mut_budget(|mut b| {
            b.enabled = false;
            let res = f();
            b.enabled = true;
            res
        })
```

Reply:
This wouldn't work because the `Budget` would be borrowed the entire time, preventing `f()` from borrowing it.

Reply:
Hmm I see, that's what I was suspecting too. But it compiled and ran passed all tests successfully when I tried. Just wanted to check.

Reply:
You brought up a good point - I was testing this witch a separate contract, but there were no test cases that used diagnostics, so I added an invocation test.